src/utility.jl

```diff
@@ -0,0 +1,13 @@
+"""
+    disable_polyester_threads(f::F)
+
+A context manager function that disables Polyester threads without affecting the scheduling
+of `Base.Treads.@threads`. Particularly useful for cases when Polyester has been used to
+multithread an inner small problem that is now to be used in an outer embarassingly parallel
+problem (in such cases it is best to multithread only at the outermost level).
+"""
+function disable_polyester_threads(f::F) where {F}
+    t, r = request_threads(num_threads())
+    f()
+    foreach(free_threads!, r)
+end
```

Comment:
Perhaps this should be
```julia
    t, r = request_threads(num_threads())
    try
        f()
    finally
        foreach(free_threads!, r)
    end
```

Also, maybe this should be in PolyesterWeave itself?
It could still be re-exported here.

Should also document that it turns off threading for `LoopVectorization.@tturbo` and `Octavian.matmul`, even though neither depend on Polyester (but they do depend on PolyesterWeave).

Do benchmark my suggestion though.
If it is slower, feel free to ignore it.

The advantage of my suggestion is that, if your code throws an error, it'll still free the threads. Otherwise, they'll be turned off until you manually call `PolyesterWeave.reset_workers!()`.
`Polyester.@batch` doesn't use `try`/`finally` either, meaning it requires the manual calls after you get errors.

Reply:
Will do:

- documenting disabling threads in tturbo and matmul
- checking whether a try block is much slower
- move to PolyesterWeave after the above is done

Reply:
After moving to PolyesterWeave, you can of course import and reexport in this PR.

Reply:
The benchmarks did not show any slowdown with a `try` block (especially and obviously if the disable command is executed once in the outermost scope).